### PR TITLE
More gradual loading for steel creep damage test 

### PR DESCRIPTION
**Pull request by recuero**
Attempt at avoiding failure of steel creep damage when using PETSc alt.

See @lindsayad's comment
https://github.com/idaholab/blackbear/pull/250#issuecomment-912611975

Damage should heavily distort finite elements when used in practical applications. I am hoping more gradual application of the loading could help in this case.

Refs #231

### CODE REVIEW

**Comment by lindsayad** [comment]
@recuero to check and see whether we're avoiding the NaNs can you remove the `method` restriction in the AD test? If we're no longer triggering the assert then we should be able to run in debug mode again

**Review by lindsayad — APPROVED** [review]
Sweet! This looks good to me. I'll give @bwspenc some time to potentially weigh-in if he wants before merging

File: test/tests/steel_creep_damage_oh/tests
Comment (bwspenc) [inline test/tests/steel_creep_damage_oh/tests:8]: Why no valgrind? Just slow?

Comment (bwspenc) [reply test/tests/steel_creep_damage_oh/tests:8]: I guess you added a bunch of time steps, right? Does the test run fast enough in opt? It should be under 2 seconds.

Comment (recuero) [reply test/tests/steel_creep_damage_oh/tests:8]: Yeah, the test takes longer with this change. I was afraid it was going to be too much. I can undo this.

**Comment by loganharbour** [comment]
Do we need to add a PETSc alt recipe to actually test this?

File: test/tests/steel_creep_damage_oh/tests
Comment (recuero) [reply test/tests/steel_creep_damage_oh/tests]: Not sure if we can have a meaningful test that runs under 2 seconds and has a more gradual loading. It takes about 10 seconds in `opt`

**Comment by loganharbour** [comment]
https://github.com/idaholab/civet_recipes/pull/78

**Comment by lindsayad** [comment]
Adding the debug method back is sufficient to test that we're avoiding the assert, and if we are it means the test will work with both versions of PETSc. But it's good to have the recipe in general

**Comment by loganharbour** [comment]
> Adding the debug method back is sufficient to test that we're avoiding the assert, and if we are it means the test will work with both versions of PETSc. But it's good to have the recipe in general

Sounds good - someone approve then when appropriate.

File: test/tests/steel_creep_damage_oh/tests
Comment (recuero) [reply test/tests/steel_creep_damage_oh/tests:8]: @bwspenc added a shorter running test that will go through valgrind, even though it doesn't get to go through the entire physics of the problem. Does that work?

Comment (bwspenc) [reply test/tests/steel_creep_damage_oh/tests:8]: Yeah, something like that should be fine. If the main test takes 10 seconds, it should be marked 'heavy'. Is the only difference between the two inputs the end time?  If so, probably best to just use cli_args. Maybe just have the standard short test with the early end time, and then a second variant with a later end time that is marked as heavy and skipped in valgrind.

Comment (recuero) [reply test/tests/steel_creep_damage_oh/tests:8]: Did that. However, as this PR stands now, it means a regression in the capabilities that are being tested. There is only `heavy` testing of the code paths and physics that account for elements actually losing load carrying capacity due to damage.

As far I can tell, tests will pass without timing out with that test not marked as `heavy`.

**Comment by recuero** [comment]
Tests are not running now. Need to wait until CIVET tests latest version before merging.

**Review by bwspenc — APPROVED** [review]

**Pull request merged**
